### PR TITLE
initial tentative to make an MPI contaier

### PR DESCRIPTION
**Pull request by mbianco**
just creating a PR to get feedback...

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -26,3 +26,5 @@ for clang_version in 9; do
     docker build --build-arg REPOSITORY=$1 --build-arg CLANG_VERSION=$clang_version -t $1:clang-$clang_version-cuda-10.1 clang-cuda
     docker build --build-arg REPOSITORY=$1 --build-arg BASE=clang-$clang_version-cuda-10.1 -t $1:test-clang-$clang_version-cuda-10.1 test
 done
+
+version=9 docker build --build-arg REPOSITORY=$1 --build-arg GCC_VERSION=$version -t $1:gcc-$version gcc
```
Comment (fthaler) [inline build.sh]: This just rebuilds the GCC 9 image from above?

File: gcc-mpi/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ARG REPOSITORY
+FROM ${REPOSITORY}:gcc
```
Comment (fthaler) [inline gcc-mpi/Dockerfile]:
```suggestion
ARG BASE
FROM ${REPOSITORY}:${BASE}
```

File: gcc-mpi/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+RUN  apt-get update && \
+     apt-get --assume-yes install openmpi-bin
```
Comment (fthaler) [inline gcc-mpi/Dockerfile]:
```suggestion
RUN  apt-get update -qq && \
     apt-get install -qq -y libopenmpi-dev && \
     rm -rf /var/lib/apt/lists/*
```

**Comment by havogt** [comment]
I'd propose we just put mpi in each of the already existing containers. What do you think @fthaler ?

**Comment by fthaler** [comment]
Yes, this could probably make sense. I will try to find out if proper container support (Sarus or Singularity) is now available or planned for Ault. If yes, we might actually move all our testing to containers and maybe also re-think the setup a bit.

**Comment by mbianco** [comment]
@fthaler would you review it?

**Comment by fthaler** [comment]
I would prefer @havogt ’s approach, that is, just install libopenmpi-dev in the base image.

**Comment by havogt** [comment]
@fthaler Somewhere you mentioned mpich would be better because of sarus, so should we just do that?

**Comment by mbianco** [comment]
I need a way to run multiple processes on a single container, and so far I could not do it with MPICH. Did you have any problem doing so?

**Comment by mbianco** [comment]
Let's see how this works. I'm not sure how I could test this myself...

**Comment by mbianco** [comment]
The mpi implementation seems to work. Could this be merged?

File: base/Dockerfile [hunk 4]
```diff
@@ -34,4 +34,8 @@ RUN cd /tmp && \
     cp -r boost_${BOOST_VERSION_UNERLINE}/boost /usr/local/include/ && \
     rm -rf *
 
+RUN  apt-get update -qq && \
+     apt-get install -qq -y mpich && \
```
Comment (fthaler) [inline base/Dockerfile:38]: Fine for me, not sure if the default version is ABI-compatible with the one installed on Daint, have you tried?

Comment (fthaler) [reply base/Dockerfile:38]: Oh and putting it to the top where the other packages are installed would improve container building performance…

Comment (mbianco) [reply base/Dockerfile:38]: nope
I'm not sure how to do it...

Comment (mbianco) [reply base/Dockerfile:38]: > Oh and putting it to the top where the other packages are installed would improve container building performance…

Not sure what this means...

Comment (fthaler) [reply base/Dockerfile:38]: > > Oh and putting it to the top where the other packages are installed would improve container building performance…
> 
> Not sure what this means...

There’s already a `apt-get install` invocation at the top of this file. If you just add mpich there, you save a round of `apt-get update` and don’t create an additional image layer.

**Pull request closed without merging**